hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaVersions.java

```diff
@@ -22,10 +22,11 @@
 
 import static java.lang.System.arraycopy;
 
+// read and updated only by partition threads
 final class PartitionReplicaVersions {
-    final int partitionId;
-    // read and updated only by operation/partition threads
-    final long[] versions = new long[InternalPartition.MAX_BACKUP_COUNT];
+    private final int partitionId;
+    private final long[] versions = new long[InternalPartition.MAX_BACKUP_COUNT];
+    private boolean dirty;
```

Comment:
Can you add a few lines of comment what 'dirty' means? It has a change which isn't replicated?